src/NServiceBus.Hosting.Windows/EndpointConfigurationExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace NServiceBus
+{
+    using Configuration.AdvanceExtensibility;
+
+    /// <summary>
+    /// Extentions for <see cref="EndpointConfiguration"/>.
+    /// </summary>
+    public static class EndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Defines the endpoint name in code thus overriding the previously assigned endpoint name.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="endpointName">The endpoint name to be used.</param>
+        public static void DefineEndpointName(this EndpointConfiguration configuration, string endpointName)
+        {
+            configuration.GetSettings().Set("NServiceBus.Routing.EndpointName", endpointName);
```

Comment:
Does this indicate we should actually use a constant defined by core?

On Monday, July 11, 2016, Daniel Marbach <daniel.marbach@particular.net>
wrote:

> Fair point. I'll add to core since the core owns it
>
> On Monday, July 11, 2016, Tim Bussmann <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
>> In src/NServiceBus.Hosting.Windows/EndpointConfigurationExtensions.cs
>> <https://github.com/Particular/NServiceBus.Host/pull/92#discussion_r70266474>
>> :
>>
>> > +{
>> > +    using Configuration.AdvanceExtensibility;
>> > +
>> > +    /// <summary>
>> > +    /// Extentions for <see cref="EndpointConfiguration"/>.
>> > +    /// </summary>
>> > +    public static class EndpointConfigurationExtensions
>> > +    {
>> > +        /// <summary>
>> > +        /// Defines the endpoint name in code thus overriding the previously assigned endpoint name.
>> > +        /// </summary>
>> > +        /// <param name="configuration">The configuration.</param>
>> > +        /// <param name="endpointName">The endpoint name to be used.</param>
>> > +        public static void DefineEndpointName(this EndpointConfiguration configuration, string endpointName)
>> > +        {
>> > +            configuration.GetSettings().Set("NServiceBus.Routing.EndpointName", endpointName);
>>
>> would it be possible to have some kind of test to make sure we will note
>> in case that config key changes? I think this could be a fairly easy unit
>> test using context.Settings.EndpointName() to verify that setting works
>> correctly?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Particular/NServiceBus.Host/pull/92/files/b5a81c38675e1a718bbd9c6b5b7df90b071105af#r70266474>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AAKostw-oz54-HknP7jTz6l2NB3D5Upcks5qUlFugaJpZM4JEWup>
>> .
>>
>